postgres/README.md

```diff
@@ -7,57 +7,124 @@
 Get metrics from PostgreSQL in real time to:
 
 - Visualize and monitor PostgreSQL states.
-- Received notifications about PostgreSQL failovers and events.
+- Receive notifications about PostgreSQL failovers and events.
+- Collect query statistics and execution plans. 
 
 ## Setup
 
-### Installation
+The Datadog Agent collects telemetry directly from the database by logging in as a read-only user.
+
+### Agent Installation
 
 The PostgreSQL check is packaged with the Agent. To start gathering your PostgreSQL metrics and logs, [install the Agent][2].
 
-### Configuration
+### Database Configuration
 
-#### Prepare Postgres
+Configure the following [Postgres parameters](https://www.postgresql.org/docs/13/config-setting.html). This is required only for [Deep Database Monitoring](#deep-database-monitoring).
 
-To get started with the PostgreSQL integration, create a read-only `datadog` user with proper access to your PostgreSQL server. Start `psql` on your PostgreSQL database.
+| Parameter | Value | Description |
+| --- | --- | --- |
+| `shared_preload_libraries` | `pg_stat_statements` | Enable collection of query metrics via the the [pg_stat_statements](https://www.postgresql.org/docs/13/pgstatstatements.html) extension. Required for `postgresql.queries.*` metrics. |
+| `track_activity_query_size` | `4096` | Increase the size of SQL text in `pg_stat_activity` and `pg_stat_statements`. Required for collection of larger queries. If left at the postgres default then queries longer than `1024` characters will not be collected. |
+| `pg_stat_statements.track` | `ALL` | Optional. Enable tracking of statements within stored procedures and functions. |
+| `pg_stat_statements.max` | `10000` | Optional. Increase the number of normalized queries tracked in `pg_stat_statements`. |
 
-For PostgreSQL version 10 and above, run:
+##### How to configure
 
-```shell
-create user datadog with password '<PASSWORD>';
-grant pg_monitor to datadog;
-grant SELECT ON pg_stat_database to datadog;
-```
+| Deployment Method | Configuration |
+| --- | --- |
+| Self-hosted | Local `postgresql.conf` file. |
+| AWS [RDS](https://aws.amazon.com/rds/postgresql/) or [RDS Aurora](https://aws.amazon.com/rds/aurora/postgresql-features/) | [DB Parameter Group](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_WorkingWithParamGroups.html) <br/>  Note that for RDS Aurora `pg_stat_statements` is enabled by default. |
+| Google [CloudSQL](https://cloud.google.com/sql) | [Database Flags](https://cloud.google.com/sql/docs/postgres/flags) |
 
-For older PostgreSQL versions, run:
+Restart the database in order for the configuration changes to take effect.
 
-```shell
-create user datadog with password '<PASSWORD>';
-grant SELECT ON pg_stat_database to datadog;
+### Agent Database Access
+
+The datadog agent requires read-only access to the database in order to collect statistics and queries.
+
```

Comment:
```suggestion
Connect to the **postgres** database and run the following SQL commands with superuser or a user with sufficient permissions to execute these commands. This can be done manually with `[psql](https://www.postgresql.org/docs/current/app-psql.html)` or using any automated configuration you use to manage your database setup.
```

Reply:
Oh yeah good point. Forgot to mention the superuser point.